### ISSUE TITLE
Ability to specify maximum number of retries?

### ISSUE THREAD

**Issue by aegershman** [concourse/pool-resource#36]
tl;dr: Is there a way to specify a "max retry" count? Or some other way to have the job fail after a certain amount of time?

My concern is that my pipelines/locks could somehow work themselves into an invalid state where a lock can _never_ be aquired, and my jobs will continue to poll-wait until the end of time. If, for example, after 12 hours of trying to aquire a lock to make changes to an Ops Manager, I'd like the job to simply fail, so that I can receive a slack notification saying the job has failed. If the lock poll-waits and never gives up, I might not know if something has gone wrong unless I go look at the pipelines myself.

Thank you for your time, I appreciate it.
